### PR TITLE
Merge changes done for the Summit review runs

### PR DESCRIPTION
**Pull request by michel2323**

### CODE REVIEW

File: src/Evaluators/ProxALEvalutor.jl [hunk 1]
```diff
@@ -27,6 +27,8 @@ function ProxALEvaluator(
     modelinfo::ModelParams,
     algparams::AlgParams,
     space::AbstractSpace=FullSpace(),
+    opt_sol::Dict = Dict(),
```
Comment (frapac) [inline src/Evaluators/ProxALEvalutor.jl:30]: I am not sure to understand why we need these two dictionaries?

File: test/blockmodel.jl [hunk 2]
```diff
@@ -105,10 +105,10 @@ load_file = joinpath(DATA_DIR, "mp_demand", "$(case)_oneweek_168")
         obj_exa = solution.minimum
         pg_exa = solution.pg
         slack_exa = solution.st
-        @test obj_jump ≈ obj_exa
-        @test pg_jump ≈ pg_exa rtol=1e-6
+        @test obj_jump ≈ obj_exa rtol=1e-3
```
Comment (frapac) [inline test/blockmodel.jl:108]: I don't understand why we need to loosen that much the tolerance. I think something is wrong

File: test/blockmodel.jl [hunk 3]
```diff
@@ -105,10 +105,10 @@ load_file = joinpath(DATA_DIR, "mp_demand", "$(case)_oneweek_168")
...
+        @test pg_jump ≈ pg_exa rtol=1e-1
         if t > 1  # slack could be of any value for t == 1
-            @test slack_jump ≈ slack_exa rtol=1e-5
+            @test slack_jump ≈ slack_exa rtol=1e-1
```
Comment (frapac) [inline test/blockmodel.jl:111]: Same here ... `rtol=1e-1` means that the test is broken...

**Review by frapac — CHANGES_REQUESTED** [review]
I think something is wrong, as we are loosening too much the tolerance of the tests, when comparing JuMP with ExaPF. We need to understand what's going on before merging this PR.

File: .github/workflows/action.yml [hunk 4]
```diff
@@ -3,24 +3,42 @@ name: Run tests
...
     steps:
-      - uses: actions/checkout@v1.0.0
-      - uses: julia-actions/setup-julia@v1
+      - uses: actions/checkout@v2
+      - uses: julia-actions/setup-julia@latest
         with:
           version: ${{ matrix.julia-version }}
+    # - run: julia --project -e 'using Pkg; Pkg.add(url="https://github.com/exanauts/Hiop.jl", rev="master")'
+      - run: julia --project -e 'using Pkg; Pkg.add(url="https://github.com/sshin23/MadNLP.jl", rev="master")'
+      - run: julia --project -e 'using Pkg; Pkg.add(url="https://github.com/exanauts/ExaPF.jl", rev="develop")'
```
Comment (frapac) [inline .github/workflows/action.yml:21]: Are we sure we want to test ProxAL with ExaPF on `develop`?

Comment (michel2323) [reply .github/workflows/action.yml:21]: Good point. I think in develop we should test with develop and on master with master? That way we aim to have the branches synchronized. It forces ProxAL to regularly update with changes in ExaPF develop.

File: src/Evaluators/ProxALEvalutor.jl (hunk 1 above)
Comment (michel2323) [reply src/Evaluators/ProxALEvalutor.jl:30]: They are used for plotting and saving the iterations. They were already there before but were moved to the Evaluator to run multiple Evaluators sequentially while adding data points. For now, I think we should leave it in to not break the plots created for the review in this commit and revisit in a later refactoring.

File: .github/workflows/action.yml (hunk 4 above)
Comment (michel2323) [reply .github/workflows/action.yml]: It's using the ExaPF release again for now.

File: test/blockmodel.jl (hunk 3 above)
Comment (michel2323) [reply test/blockmodel.jl]: Reverted to stricter tolerances.

File: test/distributed64.jl [hunk 5]
```diff
@@ -46,14 +52,53 @@ algparams.verbose = 0
...
+nlp = ProxALEvaluator(case_file, load_file, modelinfo, algparams, ProxAL.FullSpace(), opt_sol, lyapunov_sol)
 info = ProxAL.optimize!(nlp)
+nlp = ProxALEvaluator(case_file, load_file, modelinfo, algparams, ProxAL.FullSpace(), opt_sol, lyapunov_sol)
+info = ProxAL.optimize!(nlp)
+nlp = ProxALEvaluator(case_file, load_file, modelinfo, algparams, ProxAL.FullSpace(), opt_sol, lyapunov_sol)
+if MPI.Comm_rank(MPI.COMM_WORLD) == 0
+    np = MPI.Comm_size(MPI.COMM_WORLD)
+    @time info = ProxAL.optimize!(nlp)
+    println("AugLag iterations: $(info.iter) with $np ranks")
+    optimgap = 100.0abs.(info.objvalue .- zstar)/abs(zstar)
```
Comment (frapac) [inline test/distributed64.jl:88]: I would split the figure from the function: `100.0 * abs.(...`

File: test/distributed64.jl [hunk 6]
```diff
@@ -46,14 +52,53 @@ algparams.verbose = 0
...
+    lyapunov_gap = 100.0(info.lyapunov .- lyapunov_star)/abs(lyapunov_star)
```
Comment (frapac) [inline test/distributed64.jl:89]: Idem

File: .github/workflows/action.yml [hunk 7]
```diff
@@ -3,24 +3,41 @@ name: Run tests
...
+
+    steps:
       - uses: actions/checkout@v2
       - uses: julia-actions/setup-julia@latest
         with:
           version: ${{ matrix.julia-version }}
-      - run: julia --project -e 'using Pkg; Pkg.add(url="https://github.com/exanauts/Hiop.jl", rev="master")'
+    # - run: julia --project -e 'using Pkg; Pkg.add(url="https://github.com/exanauts/Hiop.jl", rev="master")'
       - run: julia --project -e 'using Pkg; Pkg.add(url="https://github.com/sshin23/MadNLP.jl", rev="master")'
+      - run: julia --project -e 'using Pkg; Pkg.add(url="https://github.com/exanauts/ExaPF.jl", rev="develop")'
```
Comment (frapac) [inline .github/workflows/action.yml:41]: I would remove this line to ensure we test ProxAL on ExaPF 0.5.

**Review by frapac — COMMENTED** [review]
I think it's almost ready to merge

File: test/distributed64.jl (hunk 5 above)
Comment (michel2323) [reply test/distributed64.jl]: Fixed.

File: test/distributed64.jl (hunk 6 above)
Comment (michel2323) [reply test/distributed64.jl]: Idem

File: .github/workflows/action.yml (hunk 7 above)
Comment (michel2323) [reply .github/workflows/action.yml]: Fixed.

**Pull request merged**